### DISCUSSION TITLE
[statusline] how can i load my custom statusline config

### DISCUSSION THREAD

**Discussion by kayuxx** [NvChad/NvChad#2209] [Q&A] [answered]
i want to customize the minimal theme in statusline and i cant load the whole config.
i saw this issue #1307 and nothing works with `overide`  property i think its no longer supported and has replaced with `overide_modules`, i tried to load the config with `run` property in `overide` and nothing works:
```lua
--custom/chadrc.lua
--....
M.ui = {
 statusline = {
  overide ={
   run = require "custom.configs.custom_statusline"
  }
 }
}
--...

```
please i want an example to customize the statusline config

**Comment by antoineco** [comment, marked as answer]
You can customize it to some extent, by overriding the existing modules.

Example:

`~/.config/nvim/lua/custom/chadrc.lua`:
```lua
---@type ChadrcConfig
local M = {}

M.ui = {
  statusline = {
    overriden_modules = function()
      return require "custom.statusline"
    end,
  },
}

return M
```

`~/.config/nvim/lua/custom/statusline.lua`:
```lua
local M = {}

local statusline = require "nvchad_ui.statusline.default"

-- add cursor column to theme's default text
M.cursor_position = function()
  local txt = statusline.cursor_position() -- " l% "
  local col = ":" .. "%c"
  local trail_i = txt:find "%s+$"
  return (trail_i and txt:sub(1, trail_i - 1) .. col .. txt:sub(trail_i)) or txt .. col
end

-- remove noisy prefix
M.LSP_status = function()
  local txt = statusline.LSP_status() -- "   LSP ~ ls_name "
  if txt then
    txt, _ = txt:gsub("LSP%s~%s", "")
    return txt
  end
end

return M
```

![image](https://github.com/NvChad/NvChad/assets/3299086/76124089-95ec-46fb-91a6-dee545982684)

![image](https://github.com/NvChad/NvChad/assets/3299086/3e068988-28b7-4b1d-9cb3-df776313a949)

**Reply by benzwt** [reply]
@antoineco 

Your solution didn't work out as of  nvchad@d3d9aa251a9dd94881c (Thu Aug 10 11:09:25 2023 +0530)
I got an error as bellow:

>...cal/share/nvim/lazy/ui/lua/nvchad/statusline/default.lua:202:... ><...cal/share/nvim/lazy/ui/lua/nvchad/statusline/default.lua:185>
>E15: Invalid expression: "v:lua.require('nvchad.statusline.default').run()"

Didn't miss something ?
thanks!

![failed](https://github.com/NvChad/NvChad/assets/1032178/1f21fe48-8491-461b-b2af-8d4416e19293)

**Reply by benzwt** [reply]
@antoineco 
@siduck had changed the nvchad_ui to nvchad.
After s/nvchad_ui/nvchad/  in statusline.lua, the error had gone. But I got not effect on the statusline.
further help is needed.

thanks!

**Reply by antoineco** [reply]
@benzwt I'd recommend creating a new discussion explaining what you are trying to do, with what statusline theme, etc. There were some potentially breaking changes in https://github.com/NvChad/ui/pull/152

**Reply by siduck** [reply]
@benzwt  https://nvchad.com/docs/config/nvchad_ui

**Comment by kayuxx** [comment]
@antoineco I rly appreciate it thank you its works for me, but I want one more thing what i can do to re order the modules for ex i want the diagnostics to be in the last order (i want to moved it to replace `cursor_postition`)

**Reply by antoineco** [reply]
For this you have to hack around the existing [modules order](https://github.com/NvChad/ui/blob/53b0d0b0e2a30e730865110eb2128dcfee7b87f6/lua/nvchad_ui/statusline/default.lua#L186-L199), and make module callbacks return something which doesn't necessarily correspond to their name:

`~/.config/nvim/lua/custom/statusline.lua`
```lua
local M = {}

local statusline = require "nvchad_ui.statusline.default"

-- swap "LSP_status" with "cursor_position"
M.cursor_position = function()
  return statusline.LSP_status()
end

-- swap "cursor_position" with "LSP_status"
M.LSP_status = function()
  return statusline.cursor_position()
end

return M
```

You can virtually achieve anything with this approach. Also, you don't have to return the exact text and style generated by the existing module, I just used these as an example. You can return something entirely custom instead.

**Comment by kayuxx** [comment]
thank you @antoineco that was very helpful
